src/black/files.py

```diff
@@ -244,6 +256,11 @@ def gen_python_files(
         if child.is_dir():
             # If gitignore is None, gitignore usage is disabled, while a Falsey
             # gitignore is when the directory doesn't have a .gitignore file.
+            if gitignore is None:
+                _gitignore = None
+            else:
+                _gitignore = gitignore.copy()
+                _gitignore[root / child] = get_gitignore(child)
```

Comment:
Can do `new_gitignore = {**gitignore, root / child: get_gitignore(child)}`.

I'd avoid using leading underscores for locals; better to give an informative name.